src/azure-cli/azure/cli/command_modules/resource/custom.py

```diff
@@ -2132,6 +2132,24 @@ def update(self, parameters):
 
     def tag(self, tags):
         resource = self.get_resource()
+
+        # please add the service type that needs to be requested with PATCH type here
+        # for example: the properties of RecoveryServices/vaults must be filled, and a PUT request that passes back
+        # to properties will fail due to the lack of properties, so the PATCH type should be used
+        need_patch_service = ['Microsoft.RecoveryServices/vaults']
+
+        if resource is not None and resource.type in need_patch_service:
```

Comment:
Can you add a test for Recovery Service tag scenario?

Reply:
Done